File: src/main/java/org/example/chessplatformbe/persistence/StreamRepository.java
package org.example.chessplatformbe.persistence;

import org.example.chessplatformbe.domain.Stream;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.Optional;

public interface StreamRepository {
    Optional<Stream> findById(Integer id);

    Stream save(Stream stream);

    void delete(Stream stream);

    Page<Stream> getStreamPage(Pageable pageable);
}

File: src/persistence/stream_repository.rs
use crate::domain::{page::Page, pageable::Pageable, stream::Stream};

pub trait StreamRepository {
    fn find_by_id(&self, id: u32) -> Option<Stream>;
    fn save(&mut self, stream: Stream) -> Stream;
    fn delete(&mut self, stream: Stream);
    fn get_stream_page(&self, pageable: Pageable) -> Page<Stream>;
}
